hpcgap/lib/fldabnum.gd

```diff
@@ -294,24 +294,28 @@ DeclareGlobalVariable( "GaussianRationals",
 DeclareCategory( "IsGaussianRationals", IsCyclotomicCollection and IsField );
 #T better?
 
+DeclareGlobalFunction( "AbelianNumberFieldByReducedGaloisStabilizerInfo" );
 
 #############################################################################
 ##
 #V  CYCLOTOMIC_FIELDS
 ##
 ##  <ManSection>
-##  <Var Name="CYCLOTOMIC_FIELDS"/>
+##  <Func Name="CYCLOTOMIC_FIELDS" Arg='n'/>
 ##
 ##  <Description>
-##  At position <A>n</A>, the <A>n</A>-th cyclotomic field is stored.
+##  Returns the <A>n</A>-th cyclotomic field.
 ##  </Description>
 ##  </ManSection>
 ##
-DeclareGlobalVariable( "CYCLOTOMIC_FIELDS",
-    "list, CYCLOTOMIC_FIELDS[n] = CF(n) if bound" );
-
-InstallFlushableValue( CYCLOTOMIC_FIELDS, [ ] );
-ShareSpecialObj(CYCLOTOMIC_FIELDS);
+BindGlobal( "CYCLOTOMIC_FIELDS",
+    MemoizePosIntFunction(
+        function(xtension)
+            return AbelianNumberFieldByReducedGaloisStabilizerInfo( Rationals,
+                   xtension, [ 1 ] );
+        end,
+    [ Rationals, Rationals,, GaussianRationals ] )
```

Comment:
For some reason this comment is only visible on the commit, not the overall PR... the problem I had/have with this is that setting the 2nd cyclotomic field happens in two files, one for HPC-GAP and once for GAP. But these two changes are introduced in two different commits